libshvcore/src/utils/shvfilejournal.cpp

```diff
@@ -757,27 +762,50 @@ chainpack::RpcValue ShvFileJournal::getLog(const ShvFileJournal::JournalContext
 						continue;
 					logDShvJournal() << "\t\t MATCH";
 				}
-				if(params_since_msec > 0 && e.epochMsec < params_since_msec) {
-					if(params.withSnapshot) {
-						if(e.sampleType == ShvJournalEntry::SampleType::Continuous) {
-							last_snapshot_msec = e.epochMsec;
-							addToSnapshot(snapshot, e);
+				this_file_values[e.path] = e;
+				entries_to_write[0] = e;
+				entries_to_write.resize(1);
+				if(rd.isInSnapShot()) {
+					prev_file_values.erase(e.path);
+				}
+				else if(!prev_file_values.empty()) {
+					for(const auto &kv : prev_file_values) {
+						ShvJournalEntry e3 = kv.second;
+						if(!e3.value.hasDefaultValue() && this_file_values.find(e3.path) == this_file_values.cend()) {
```

Comment:
Ta druha cast podminky je zbytecna, ne? Pokud se to dostalo do `this_file_values` tak uz je to vymazane z `prev_file_values`? Nechci byt puntickar, pokud to tam mas pro jistotu, tak mi to nevadi, jen jestli to spravne chapu.

Reply:
jj

Reply:
opraveno 3393e66